wordpress-mobile/WordPress-iOS
# Add supported block parameter to fetch page layout call

**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2444

**Related PRs:**
`gutenberg:` https://github.com/WordPress/gutenberg/pull/24976
`gutenberg-mobile:` https://github.com/wordpress-mobile/gutenberg-mobile/pull/2589

## To test:
<details>
<summary><strong>To disable or enable the development version of Modal Layout Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <img width=300 src="https://user-images.githubusercontent.com/3384451/90816133-c8b10580-e2f9-11ea-8584-6bbfd7e523e2.gif" />
</details>

When enabled, the Layout Picker should show when creating a new page from My Site and from the Page list.

### WPCom Site
1. Select a WPcom site and ensure a valid network connection
1. Navigate to the Modal layout picker by either:
    - Selecting the FAB icon on the site page and selecting "Site page"
    - Selecting the ➕ on the Page List
1. Using an earlier build from at least https://github.com/wordpress-mobile/WordPress-iOS/pull/14726 or later validate the number of layouts has decreased
1. Using a network logger:
    - **Expect** to see a network request with: 
```
https://public-api.wordpress.com/wpcom/v2/sites/{{SITEID}}/block-layouts?locale=en&supported_blocks=core/paragraph%2Ccore/heading%2Ccore/more%2Ccore/image%2Ccore/video%2Ccore/nextpage%2Ccore/separator%2Ccore/list%2Ccore/quote%2Ccore/media-text%2Ccore/preformatted%2Ccore/gallery%2Ccore/columns%2Ccore/column%2Ccore/group%2Ccore/freeform%2Ccore/button%2Ccore/spacer%2Ccore/shortcode%2Ccore/buttons%2Ccore/latest-posts%2Ccore/verse%2Ccore/cover%2Ccore/social-link%2Ccore/social-links%2Cjetpack/contact-info%2Cjetpack/email%2Cjetpack/phone%2Cjetpack/address%2Ccore/pullquote%2Ccore/code
```

### Self-Hosted Site
1. Select a Self-Hosted site and ensure a valid network connection
1. Navigate to the Modal layout picker by either:
    - Selecting the FAB icon on the site page and selecting "Site page"
    - Selecting the ➕ on the Page List
1. Using an earlier build from at least https://github.com/wordpress-mobile/WordPress-iOS/pull/14726 or later validate the number of layouts has decreased
1. Using a network logger:
    - **Expect** to see a network request with: 
```
https://public-api.wordpress.com/wpcom/v2/common-block-layouts?locale=en&supported_blocks=core/paragraph%2Ccore/heading%2Ccore/more%2Ccore/image%2Ccore/video%2Ccore/nextpage%2Ccore/separator%2Ccore/list%2Ccore/quote%2Ccore/media-text%2Ccore/preformatted%2Ccore/gallery%2Ccore/columns%2Ccore/column%2Ccore/group%2Ccore/freeform%2Ccore/button%2Ccore/spacer%2Ccore/shortcode%2Ccore/buttons%2Ccore/latest-posts%2Ccore/verse%2Ccore/cover%2Ccore/social-link%2Ccore/social-links%2Cjetpack/contact-info%2Cjetpack/email%2Cjetpack/phone%2Cjetpack/address%2Ccore/pullquote%2Ccore/code
```
## Screenshots:
<img width="300" alt="Screen Shot 2020-08-27 at 5 17 58 PM" src="https://user-images.githubusercontent.com/3384451/91495909-4a200f00-e889-11ea-959d-c7b05a09a1b2.png">

## Additional Context: 

This is a continuation of the work done in https://github.com/wordpress-mobile/WordPress-iOS/pull/14726 and will be followed up by other tickets as part of [Modal Layout Picker Project](https://github.com/wordpress-mobile/gutenberg-mobile/issues?q=is%3Aissue+is%3Aopen+%5BModal+Layout+Picker%5D)

## PR submission checklist:
